app/screens/edit_channel/edit_channel.js

```diff
@@ -0,0 +1,297 @@
+
+        const cleanedName = cleanUpUrlable(channelURL);
+        if (cleanedName === channelURL) {
+            return {error: null};
+        }
+
+        return {error: formatMessage(messages.name_lowercase)};
+    };
+
+    onUpdateChannel = () => {
+        Keyboard.dismiss();
+        const {displayName, channelURL, purpose, header} = this.state;
+        const {channel: {id}} = this.props;
+        const channel = {
+            display_name: displayName,
+            name: channelURL,
+            purpose,
+            header
+        };
+
+        let result = this.validateDisplayName(displayName.trim());
+        if (result.error) {
+            this.setState({error: result.error});
+            return;
+        }
+        result = this.validateChannelURL(channelURL.trim());
+        if (result.error) {
+            this.setState({error: result.error});
+            return;
+        }
+
+        this.props.actions.patchChannel(id, channel);
+    };
+
+    onNavigatorEvent = (event) => {
+        if (event.type === 'NavBarButtonPress') {
+            switch (event.id) {
+            case 'close-edit-channel':
+                this.close();
+                break;
+            case 'edit-channel':
+                this.onUpdateChannel();
+                break;
+            }
+        }
+    };
+
+    onDisplayNameChange = (displayName) => {
+        this.setState({displayName});
+    };
+
+    onChannelURLChange = (channelURL) => {
+        this.setState({channelURL});
+    };
+
+    onPurposeChange = (purpose) => {
+        this.setState({purpose});
+    };
+
+    onHeaderChange = (header) => {
+        this.setState({header});
+    };
+
+    render() {
+        const {
+            channel: {
+                display_name: oldDisplayName,
+                name: oldChannelURL,
+                header: oldHeader,
+                purpose: oldPurpose
+            },
+            navigator,
+            theme,
+            currentTeamUrl,
+            isDMorGM
+        } = this.props;
+        const {
+            error,
+            updating,
+            displayName,
+            channelURL,
+            purpose,
+            header
+        } = this.state;
+
+        return (
+            <ChannelInfo
+                navigator={navigator}
+                theme={theme}
+                enableRightButton={this.emitCanUpdateChannel}
+                error={error}
+                updating={updating}
+                displayHeaderOnly={isDMorGM}
+                currentTeamUrl={currentTeamUrl}
+                onDisplayNameChange={this.onDisplayNameChange}
+                onChannelURLChange={this.onChannelURLChange}
+                onPurposeChange={this.onPurposeChange}
+                onHeaderChange={this.onHeaderChange}
+                displayName={displayName}
+                channelURL={channelURL}
+                header={header}
+                purpose={purpose}
+                editing={true}
+                oldChannel={{
```

Comment:
Can you pass these in 1 by 1? As it is, we're passing a new object on each render, even if nothing changed